cql3/user_types.cc

```diff
@@ -295,4 +295,25 @@ void user_types::setter::execute(mutation& m, const clustering_key_prefix& row_k
     }
 }
 
+void user_types::setter_by_field::execute(mutation& m, const clustering_key_prefix& row_key, const update_parameters& params) {
+    assert(column.type->is_user_type() && column.type->is_multi_cell());
+
+    auto value = _t->bind(params._options);
+    if (value == constants::UNSET_VALUE) {
+        return;
+    }
+
+    auto& type = static_cast<const user_type_impl&>(*column.type);
+
+    auto idx = type.idx_of_field(_field_name);
+    assert(idx && *idx < type.size());
+
+    collection_mutation_description mut;
+    mut.cells.emplace_back(serialize_field_index(*idx), value
+                ? params.make_cell(*type.type(*idx), *value->get(params._options), atomic_cell::collection_member::yes)
+                : make_dead_cell(params));
+
+    m.set_cell(row_key, column, mut.serialize(type));
+}
```

Comment:
set_cell() is misnamed, it merges the input into m, or this would behave badly when we set two fields in the same column.